scrapers/bugzilla.py

```diff
@@ -0,0 +1,108 @@
+# SPDX-License-Identifier: GPL-3.0+
+from __future__ import unicode_literals
+
+from builtins import bytes
+
+from scrapers.base import BaseScraper
+from purview.models.bugzilla import BugzillaBug
+from purview.models.user import User
+import purview.utils.general as utils
+from purview import log
+
+
+class BugzillaScraper(BaseScraper):
+
+    def run(self, since=None):
+        log.info('Starting initial load of Bugzilla bugs')
+        # Initialize a start date from which all builds must be fetched
+        # If no input is given by the user, fetch builds from the past two years
+        if since is None:
+            since = self.default_since
+        else:
+            since = utils.timestamp_to_datetime(since)
+
+        bugs = self.get_bugzilla_bugs(since)
+        log.info('Successfully fetched {0} bugs from teiid'.format(len(bugs)))
+        self.update_neo4j(bugs)
+        log.info('Initial load of Bugzilla bugs complete!')
+
+    def get_bugzilla_bugs(self, start_date):
+        # SQL query to fetch all (Red Hat) bugs from start date until now
+        log.info('Getting all Bugzilla bugs since {0}'.format(start_date))
+        sql_query = """
+            SELECT bugs.*, products.name AS product_name, classifications.name AS classification,
+                assigned.login_name AS assigned_to_email, assigned.realname AS assigned_to_name,
+                reported.login_name AS reported_by_email, reported.realname AS reported_by_name,
+                qa.login_name AS qa_contact_email, qa.realname AS qa_contact_name
+            FROM bugzilla.bugs AS bugs
+            LEFT JOIN bugzilla.products AS products ON bugs.product_id = products.id
+            LEFT JOIN bugzilla.classifications AS classifications
+                ON products.classification_id = classifications.id
+            LEFT JOIN bugzilla.profiles AS assigned ON bugs.assigned_to = assigned.userid
+            LEFT JOIN bugzilla.profiles AS reported ON bugs.reporter = reported.userid
+            LEFT JOIN bugzilla.profiles AS qa ON bugs.qa_contact = qa.userid
+            WHERE classifications.name = 'Red Hat' AND bugs.creation_ts > '{}' AND bugs.reporter = 359538
+            """.format(start_date)
+
+        return self.teiid.query(sql=sql_query, retry=3)
+
+    def create_user_node(self, email, name):
+        # if email is a Red Hat email address, username is
+        # same as domain name prefix in the email address
+        # else store email as username
+        if email.split('@')[1] == 'redhat.com':
+            username = email.split('@')[0]
+        else:
+            username = email
+
+        user = User.create_or_update(dict(
+            username=username,
+            name=bytes(name, 'utf-8').decode(),
```

Comment:
Let's remove the name here since this isn't the canonical source for names (it's a free-form setting field).